lib/api/objectGet.js

```diff
@@ -154,14 +154,18 @@ function objectGet(authInfo, request, returnTagCount, log, callback) {
                 }
             }
             if (partNumber) {
-                const partObj = objMD.location[partNumber - 1];
-                if (partObj === undefined) {
+                // Get all parts that belong to the requested part number
+                const locationParts = objMD.location.filter(location =>
+                    location.dataStoreETag.startsWith(`${partNumber}`));
```

Comment:
you shoud consider adding a trailing colon in the string to match. e.g.
```location.dataStoreETag.startsWith(`${partNumber}:`));```

In addition, this must be compatible with object with no ETag in the location array